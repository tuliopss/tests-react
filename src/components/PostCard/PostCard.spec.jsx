import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import PostCard from "./PostCard";

const props = {
  title: "title",
  body: "body",
  id: "1",
  cover: "img/src.png",
};

describe("<PostCard/>", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: props.title })).toHaveAttribute(
      "src",
      props.cover
    );

    expect(
      screen.getByRole("heading", { name: /title 1/i })
    ).toBeInTheDocument();

    expect(screen.getByText("body")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
