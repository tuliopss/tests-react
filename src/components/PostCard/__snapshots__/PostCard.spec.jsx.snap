// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PostCard/> > should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title"
    src="img/src.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title
       
      1
    </h2>
    <p>
      body
    </p>
  </div>
</div>
`;
